### PR TITLE
dev-go/delve: various fixes and bump to new version

### PR DESCRIPTION
**Pull request by scardracs**
Closes: https://bugs.gentoo.org/722690
Fixed test

Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Marco Scardovi <marco@scardovi.com>

### CODE REVIEW

File: dev-go/delve/delve-1.6.0.ebuild [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
-RESTRICT="test"
+DEPEND="dev-lang/go:="
 
 src_compile() {
-	go build -ldflags="-X main.Build=6c8c4508db7cbd735354e274c17ccecca9a8501a" -o "${S}/dlv" ./cmd/dlv || die
+	go build -mod vendor -ldflags="-X main.Build=75485427957fba167a6eaca76b1ee9549d62916f" -o "${S}/dlv" ./cmd/dlv || die
 }
 
+src_test() {
+	go test
```
Comment (scardracs) [inline dev-go/delve/delve-1.6.0.ebuild:28]: Testing only with that works, do you think is necessary something else?

File: dev-go/delve/delve-1.6.0.ebuild [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
 DESCRIPTION="A source-level debugger for the Go programming language"
 HOMEPAGE="https://github.com/go-delve/delve"
 
 LICENSE="MIT BSD BSD-2 Apache-2.0"
 SLOT="0"
+IUSE="test"
+RESTRICT="!test? ( test )"
 
-RESTRICT="test"
+DEPEND="dev-lang/go:="
```
Comment (scardracs) [inline dev-go/delve/delve-1.6.0.ebuild:21]: Actually delve is a go debugger so I thought is quite obvious have it as depend for both build an and functioning

**Comment by EdoaLive** [comment]
Tested the ebuild, I confirm it works and solves https://bugs.gentoo.org/722690

File: dev-go/delve/delve-1.6.0.ebuild (hunk 2 above)
Comment (juippis) [reply dev-go/delve/delve-1.6.0.ebuild:21]: This is handled by *go-module.eclass* anyway.

File: dev-go/delve/delve-1.6.0.ebuild (hunk 1 above)
Comment (juippis) [reply dev-go/delve/delve-1.6.0.ebuild:28]: The test phase does nothing to me, 
```
>>> Test phase: dev-go/delve-1.6.0
# get https://proxy.golang.org/gopkg.in/yaml.v2/@v/v2.2.1.mod
# get https://proxy.golang.org/github.com/spf13/cobra/@v/v0.0.0-20170417170307-            b6cb39589372.mod
# get https://proxy.golang.org/github.com/creack/pty/@v/v1.1.9.mod
# get https://proxy.golang.org/github.com/google/go-dap/@v/v0.4.0.mod
# get https://proxy.golang.org/golang.org/x/tools/@v/v0.0.0-20191127201027-ecd32218bd7f.mod
# get https://proxy.golang.org/github.com/peterh/liner/@v/v0.0.0-20170317030525-           88609521dc4b.mod
# get https://proxy.golang.org/github.com/cpuguy83/go-md2man/@v/v1.0.10.mod
# get https://proxy.golang.org/go.starlark.net/@v/v0.0.0-20200821142938-949cc6f4b097.mod
# get https://proxy.golang.org/github.com/spf13/pflag/@v/v0.0.0-20170417173400-            9e4c21054fa1.mod
# get https://proxy.golang.org/go.starlark.net/@v/v0.0.0-20200821142938-949cc6f4b097.mod:  Get "https://proxy.golang.org/go.starlark.net/@v/v0.0.0-20200821142938-949cc6f4b097.mod":  dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect:     network is unreachable
# get https://proxy.golang.org/github.com/google/go-dap/@v/v0.4.0.mod: Get "https://proxy. golang.org/github.com/google/go-dap/@v/v0.4.0.mod": dial tcp: lookup proxy.golang.org on   192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/golang.org/x/sys/@v/v0.0.0-20200625212154-ddb9806d33ae.mod
# get https://proxy.golang.org/github.com/sirupsen/logrus/@v/v1.6.0.mod
# get https://proxy.golang.org/github.com/cosiner/argv/@v/v0.1.0.mod
# get https://proxy.golang.org/gopkg.in/yaml.v2/@v/v2.2.1.mod: Get "https://proxy.golang.  org/gopkg.in/yaml.v2/@v/v2.2.1.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53:  dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/cpuguy83/go-md2man/@v/v1.0.10.mod: Get "https:// proxy.golang.org/github.com/cpuguy83/go-md2man/@v/v1.0.10.mod": dial tcp: lookup proxy.    golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/peterh/liner/@v/v0.0.0-20170317030525-           88609521dc4b.mod: Get "https://proxy.golang.org/github.com/peterh/liner/@v/v0.0.0-         20170317030525-88609521dc4b.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53:     dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/mattn/go-colorable/@v/v0.0.0-20170327083344-     ded68f7a9561.mod
# get https://proxy.golang.org/github.com/spf13/cobra/@v/v0.0.0-20170417170307-            b6cb39589372.mod: Get "https://proxy.golang.org/github.com/spf13/cobra/@v/v0.0.0-          20170417170307-b6cb39589372.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53:     dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/creack/pty/@v/v1.1.9.mod: Get "https://proxy.    golang.org/github.com/creack/pty/@v/v1.1.9.mod": dial tcp: lookup proxy.golang.org on 192. 168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/hashicorp/golang-lru/@v/v0.5.4.mod
# get https://proxy.golang.org/golang.org/x/arch/@v/v0.0.0-20190927153633-4e8777c89be4.mod
# get https://proxy.golang.org/github.com/inconshreveable/mousetrap/@v/v1.0.0.mod
# get https://proxy.golang.org/github.com/mattn/go-isatty/@v/v0.0.3.mod
# get https://proxy.golang.org/golang.org/x/tools/@v/v0.0.0-20191127201027-ecd32218bd7f.   mod: Get "https://proxy.golang.org/golang.org/x/tools/@v/v0.0.0-20191127201027-            ecd32218bd7f.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/mattn/go-isatty/@v/v0.0.3.mod: Get "https://     proxy.golang.org/github.com/mattn/go-isatty/@v/v0.0.3.mod": dial tcp: lookup proxy.golang. org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/mattn/go-colorable/@v/v0.0.0-20170327083344-     ded68f7a9561.mod: Get "https://proxy.golang.org/github.com/mattn/go-colorable/@v/v0.0.0-   20170327083344-ded68f7a9561.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53:     dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/golang.org/x/sys/@v/v0.0.0-20200625212154-ddb9806d33ae.mod: Get "https://proxy.golang.org/golang.org/x/sys/@v/v0.0.0-20200625212154-ddb9806d33ae.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect:     network is unreachable
# get https://proxy.golang.org/golang.org/x/arch/@v/v0.0.0-20190927153633-4e8777c89be4.    mod: Get "https://proxy.golang.org/golang.org/x/arch/@v/v0.0.0-20190927153633-4e8777c89be4.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53:        connect: network is unreachable
# get https://proxy.golang.org/github.com/inconshreveable/mousetrap/@v/v1.0.0.mod: Get     "https://proxy.golang.org/github.com/inconshreveable/mousetrap/@v/v1.0.0.mod": dial tcp:   lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is    unreachable
# get https://proxy.golang.org/github.com/hashicorp/golang-lru/@v/v0.5.4.mod: Get "https://proxy.golang.org/github.com/hashicorp/golang-lru/@v/v0.5.4.mod": dial tcp: lookup proxy.   golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/sirupsen/logrus/@v/v1.6.0.mod: Get "https://     proxy.golang.org/github.com/sirupsen/logrus/@v/v1.6.0.mod": dial tcp: lookup proxy.golang. org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/spf13/pflag/@v/v0.0.0-20170417173400-            9e4c21054fa1.mod: Get "https://proxy.golang.org/github.com/spf13/pflag/@v/v0.0.0-          20170417173400-9e4c21054fa1.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53:     dial udp 192.168.1.1:53: connect: network is unreachable
# get https://proxy.golang.org/github.com/cosiner/argv/@v/v0.1.0.mod: Get "https://proxy.  golang.org/github.com/cosiner/argv/@v/v0.1.0.mod": dial tcp: lookup proxy.golang.org on    192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
go: github.com/cosiner/argv@v0.1.0: Get "https://proxy.golang.org/github.com/cosiner/argv/@v/v0.1.0.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:  53: connect: network is unreachable
>>> Completed testing dev-go/delve-1.6.0
```
You need to download the modules prior to test phase, it uses different / extra modules to those needed to build the program.

**Review by juippis — APPROVED** [review]

File: dev-go/delve/delve-1.6.0.ebuild (hunk 1 above)
Comment (scardracs) [reply dev-go/delve/delve-1.6.0.ebuild:28]: Can I set an additional SRC_URI if test is used?

Comment (juippis) [reply dev-go/delve/delve-1.6.0.ebuild:28]: I do not know how it's going to work with *go-module.eclass*, but in theory, yes.

Comment (scardracs) [reply dev-go/delve/delve-1.6.0.ebuild:28]: I will take a look on it, thanks :)

**Pull request closed without merging**